spec.md

```diff
@@ -97,4507 +77,377 @@ The following is an incomplete list of features, discussed during the process of
 
-Classically, repository names have always been two path components where each path component is less than 30 characters.
-The V2 registry API does not enforce this.
-The rules for a repository name are as follows:
+### Official Certification
 
-1. A repository name is broken up into _path components_.
-A component of a repository name MUST begin with one or more lowercase alpha-numeric characters. Subsequent lowercase alpha-numeric characters are OPTIONAL and MAY be separated by periods, dashes or underscores.
-More strictly, it MUST match the regular expression `[a-z0-9]+(?:[._-][a-z0-9]+)*`.
-2. If a repository name has two or more path components, they MUST be separated by a forward slash ("/").
-3. The total length of a repository name, including slashes, MUST be less than 256 characters.
+Registry providers can self-certify by submitting conformance results to [opencontainers/oci-conformance](https://github.com/opencontainers/oci-conformance).
 
-These name requirements _only_ apply to the registry API and SHOULD accept a superset of what is supported by other components.
+### Workflow Categories
 
-All endpoints SHOULD support aggressive http caching, compression and range headers, where appropriate.
-The new API attempts to leverage HTTP semantics where possible but MAY break from standards to implement targeted features.
+#### Pull
+The process of pulling an image centers around retrieving these two components. An "image" is a combination of a JSON
+manifest and individual layer files.
 
-For detail on individual endpoints, please see the [_Detail_](#detail) section.
+The first step in pulling an image is to retrieve the manifest.
 
-### Errors
+##### Pulling manifests
 
-Actionable failure conditions, covered in detail in their relevant sections, are reported as part of 4xx responses, in a json response body.
-One or more errors will be returned in the following format:
+To pull a manifest, perform a `GET` request to a url in the following form:
+[3a](#Endpoints) `/v2/<name>/manifests/<reference>`
 
-```json
-    {
-        "errors": [
-            {
-                "code": "<error identifier>",
-                "message": "<message describing condition>",
-                "detail": "<unstructured>"
-            },
-            ...
-        ]
-    }
-```
+`<name>` refers to the namespace of the repository. `<reference>` MUST be either (a) the digest of the manifest or (b) a tag name.
+The `<reference>` MUST NOT be in any other format.
 
-The `code` field will be a unique identifier, all caps with underscores by convention.
-The `message` field will be a human readable string.
-The OPTIONAL `detail` field MAY contain arbitrary json data providing information the client can use to resolve the issue.
+A GET request to an existing manifest URL MUST provide the expected manifest, with a response code that MUST be `200 OK`.
```

Comment:
expected.. is pretty broad... before ending this section should make a statement regarding how the client should use the response to validate the manifest .. and/or link to the detail doc for the associated APIs.

Is the idea that the reader will hop back and forth between this doc and detail? Or is this document supposed to stand on it's own?

This document seems to be more of an outline now ... with an intro to the workflows. We should probably not have MUST language in the rough description of each flow.  Otherwise the implementor will feel like they have to check two places with two different descriptions of the APIs to see what they MUST implement for each.

Reply:
We feel the document stands on its own (or should).

Regarding validation of manifest etc., that seems to be something that should be discussed in the spec which defines it (image spec).

We were ready to do without the details section, but added it per maintainers' request. Ultimately the conformance section should be everything needed to implement the spec.

Reply:
>  "stands on its own (or should)" ... "conformance section should be everything needed to implement the spec"

Given the answer to the image digest/signature, I presume you mean with respect to the details document that you don't want there, not with respect to other OCI spec docs. I'm against having everything needed to implement the distribution spec APIs in "two" separate documents, esp. where the MUST language of the APIs is concerned.

> "should be discussed in the spec which defines it (image spec)" 

The OCI image spec does define this as a MUST requirement and further specifies that "retrieved content SHOULD be verified against this digest when consumed via untrusted sources."  I should hope the same will be true for all artifact types.  We are talking about the "when consumed part."  What do you think about using some general language here, for all the workflows if you like, for suggesting that "verify" step(s) MUST be provided according to the requirements and suggestions made by the specifications for the media type being distributed, IE for OCI manifests, blobs.. etc.. A simple example can be included for pull, but I would agree you don't have to say the same thing on every workflow/ every API.

Reply:
summary resolution:
TODO: add general text about artifact validation requirements 

goal will be to make spec.md standalone from details.md and we me be able to remove most of the contents of details.md